commit aee422b0b2c4e54f5cec46d965e88fb8eefde6bb
Author: Ahmad Samir <a.samirh78@gmail.com>
Date:   Sat Nov 30 18:34:53 2019 +0200

    Replace KRecursiveFilterProxyModel with QSortFilterProxyModel
    
    Summary:
    KRecursiveFilterProxyModel was only needed to have recursive filtering, which
    became available upstream since Qt 5.10.  Also increate min. required Qt version
    to 5.10.
    
    Test Plan: The code builds, and recursive filtering still works
    
    Reviewers: #ark, elvisangelaccio, apol
    
    Reviewed By: apol
    
    Subscribers: apol, kde-frameworks-devel, kde-utils-devel
    
    Tags: #ark
    
    Maniphest Tasks: T12269
    
    Differential Revision: https://phabricator.kde.org/D25631

diff --git b/CMakeLists.txt a/CMakeLists.txt
index bf60f0e4..29828b44 100644
--- b/CMakeLists.txt
+++ a/CMakeLists.txt
@@ -7,7 +7,7 @@ set (KDE_APPLICATIONS_VERSION_MICRO "70")
 set (KDE_APPLICATIONS_VERSION "${KDE_APPLICATIONS_VERSION_MAJOR}.${KDE_APPLICATIONS_VERSION_MINOR}.${KDE_APPLICATIONS_VERSION_MICRO}")
 project(ark VERSION ${KDE_APPLICATIONS_VERSION})
 
-set(QT_MIN_VERSION 5.10.0)
+set(QT_MIN_VERSION 5.8.0)
 set(KF5_MIN_VERSION 5.44.0)
 
 find_package(ECM ${KF5_MIN_VERSION} REQUIRED NO_MODULE)
diff --git b/part/archivesortfiltermodel.cpp a/part/archivesortfiltermodel.cpp
index 2116b26b..2d5a92c1 100644
--- b/part/archivesortfiltermodel.cpp
+++ a/part/archivesortfiltermodel.cpp
@@ -26,10 +26,8 @@
 using namespace Kerfuffle;
 
 ArchiveSortFilterModel::ArchiveSortFilterModel(QObject *parent)
-    : QSortFilterProxyModel(parent)
+    : KRecursiveFilterProxyModel(parent)
 {
-    // always enable recursive fitlering
-    setRecursiveFilteringEnabled(true);
 }
 
 ArchiveSortFilterModel::~ArchiveSortFilterModel()
diff --git b/part/archivesortfiltermodel.h a/part/archivesortfiltermodel.h
index 9e3842fd..751f1874 100644
--- b/part/archivesortfiltermodel.h
+++ a/part/archivesortfiltermodel.h
@@ -21,9 +21,9 @@
 #ifndef ARCHIVESORTFILTERMODEL_H
 #define ARCHIVESORTFILTERMODEL_H
 
-#include <QSortFilterProxyModel>
+#include <KRecursiveFilterProxyModel>
 
-class ArchiveSortFilterModel: public QSortFilterProxyModel
+class ArchiveSortFilterModel: public KRecursiveFilterProxyModel
 {
     Q_OBJECT
 
